api/v1beta1/policyautomation_types.go

```diff
@@ -52,6 +54,25 @@ type AutomationDef struct {
 	TowerSecret string `json:"secret"`
 	// JobTTL sets the time to live for the Kubernetes AnsibleJob object after the Ansible job run has finished.
 	JobTTL *int `json:"jobTtl,omitempty"`
+	// The maximum number of violation contexts will be provided to the Ansible Tower as extra variable.
+	// +kubebuilder:validation:Minimum=0
```

Comment:
@mprahl  Both `the user set PolicyViolationContextLimit to 0` and `the use doesn't set PolicyViolationContextLimit` cases will use the default value 1000 since `unset is also zero here`. We can use other default values than 1000 or remove the limitation.  But I'm not sure if there is a need or a good way to treat `PolicyViolationContextLimit = null` and `PolicyViolationContextLimit = 0` as different cases.

Reply:
Could we use a default here with the `+kubebuilder:default` marker? Then we could consider having 0 be considered no limit.

Reply:
@mprahl So there are two solutions:

1）When the user set `PolicyViolationContextLimit = 0` or doesn‘t set `PolicyViolationContextLimit` (the default value is `zero`),  there isn't limitation.

2) When the user set `PolicyViolationContextLimit = 0` or doesn‘t set `PolicyViolationContextLimit` (the default value is `zero`), then use the default limitation = 1000.

Reply:
I think the behavior that we want is the user explicitly specifying 0 causes there to be no limit, otherwise, it'll default to `1000` or perhaps a smaller number.